src/site/topic16.rst

```diff
@@ -34,10 +34,40 @@ Topic #16 --- Iterators
 Iterators
 =========
 
+* Iterators are objects that allow us to *iterate* over a collection, one element at a time
+    * Get each element in an array
+    * Get all the elements from a bag
+
+* There are two important interfaces:
+    * ``Iterator`` --- used when creating an iterator object
+    * ``Iterable`` --- used when creating some collection that we want to be iterable
```

Comment:
Not necessarily a collection, rather anything that you want to be able to iterator over. Could be a byte stream, who knows.